flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceOptions.java

```diff
@@ -236,6 +230,8 @@ private PulsarSourceOptions() {
                                             " This argument is required when constructing the consumer.")
                                     .build());
 
+    /** @deprecated This config option is no longer supported. */
+    @Deprecated
     public static final ConfigOption<SubscriptionType> PULSAR_SUBSCRIPTION_TYPE =
```

Comment:
ditto

Reply:
This configure option will be auto remove in generated documentation. And I think we can remove it in the next release as above.

Reply:
When you refer to "the next release", do you mean the next Flink release, or the next Flink Pulsar Connector release? This patch will be delivered to the next Flink Pulsar Connector release exactly.

Reply:
I mean the next 1.18 flink release. We should inform the user first this method is no longer supported. Then we can remove it. This PR also remove some deprecated methods and config options which was deprecated in 1.16 release.